extra/release_tool.py

```diff
@@ -1592,39 +1639,47 @@ def trigger_build(state, tag_avail):
                 state.update(yaml.safe_load(fd))
             # Trigger update of parameters from disk.
             params = None
-            continue
-        elif not reply.upper().startswith("Y"):
-            return
 
-        substr = ask("Which one (substring is ok as long as it's unique)? ")
-        found = 0
-        for param in params.keys():
-            if param == substr:
-                # Exact match
-                name = param
-                found = 1
-                break
-            if param.find(substr) >= 0:
-                name = param
-                found += 1
-        if found == 0:
-            print("Parameter not found!")
-            continue
-        elif found > 1:
-            print("String not unique!")
-            continue
-        params[name] = ask("Ok. New value? ")
+        elif reply.upper() == "R":
+            extra_buildparams = None
+            params = None
+            update_state(state, ["extra_buildparams"], {})
 
-        if extra_buildparams.get(name) is not None:
-            # Extra build parameters, that are not part of the build tags for
-            # each repository, should be saved persistently in the state file so
-            # that they can be repeated in subsequent builds.
-            update_state(state, ["extra_buildparams", name], params[name])
+        elif reply.upper() == "ET" or reply.upper() == "DT":
+            if reply.upper().startswith("E"):
+                action = "true"
+                set_param("BUILD_CLIENT", action)
+                set_param("BUILD_SERVERS", action)
+            else:
+                action = "false"
+            set_param("RUN_INTEGRATION_TESTS", action)
+            for param in params.keys():
+                if param.startswith("TEST_"):
+                    set_param(param, action)
+                    build_param = re.sub("^TEST_", "BUILD_", param)
+                    if action == "true" and params.get(build_param) is not None:
+                        set_param(build_param, action)
+
+        elif reply.upper() == "EP" or reply.upper() == "DP":
+            if reply.upper().startswith("E"):
+                action = "true"
+            else:
+                action = "false"
+            set_param("BUILD_CLIENT", action)
+            set_param("BUILD_SERVERS", action)
```

Comment:
I don't think that "disable platforms" should also disable `BUILD_CLIENT` and `BUILD_SERVERS`. Which is the use case?

I would say that "disable platforms" refers to "disable build on everything not strictly needed for a release", leaving `BUILD_CLIENT` and `BUILD_SERVERS`, which are required for publishing the release. In other words, this code here could look something like:

```suggestion
        elif reply.upper() == "EP" or reply.upper() == "DP":
            if reply.upper().startswith("E"):
                action = "true"
                set_param("BUILD_CLIENT", action)
                set_param("BUILD_SERVERS", action)
            else:
                action = "false"
```

Thinking about it from another angle, this way would complement the "enable tests" case above, where "disable tests" does not touch BUILD_CLIENT, BUILD_SERVERS

Reply:
> I don't think that "disable platforms" should also disable `BUILD_CLIENT` and `BUILD_SERVERS`. Which is the use case?

I guess I'm looking at it more from the test perspective. It's quite a common use case for me. If I need to debug something on a runner: Disable all platforms and then enable the one I want. In this scenario, `BUILD_CLIENT` and `BUILD_SERVERS` are unneeded.

But you are right that from a release perspective it doesn't make sense. How about renaming it to "build combinations" instead of "platforms"? And then when enabling the release publishing, or all tests, it will turn those two back on again?

Reply:
Resolved according to discussion in Slack.